### PR TITLE
docs(tutorials): add Setup OpenTelemetry tutorial

### LINKED ISSUE #20155: 🚀 Feature: [Docs] Guide for setting up and using OpenTelemetry metrics and traces
### 🔖 Feature description

As there are more and more metrics and traces being setup in Backstage it would be really helful to have a guide that shows you how to set up and use OpenTelemetry

### 🎤 Context

We'd like to use the metrics and traces that have been added but it feels very open as to how one would set things up and start using them

### ✌️ Possible Implementation

This could be as simple as a Contrib guide like the one for Prometheus here: https://github.com/backstage/backstage/blob/master/contrib/docs/tutorials/prometheus-metrics.md

But it would be even better if it was part of the main Documentation under the Tutorials heading here: https://github.com/backstage/backstage/tree/master/docs/tutorials

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

### PR DESCRIPTION
**Pull request by tcardonne**
## Hey, I just made a Pull Request!

Adds a simple tutorial to setup OpenTelemetry in a Backstage project using simple Console exporters.

This should close #20155. Tell me if you see something's missing!

#### :heavy_check_mark: Checklist

- [ ] ~A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))~
- [X] Added or updated documentation
- [ ] ~Tests for new functionality and regression tests for bug fixes~
- [ ] ~Screenshots attached (for UI changes)~
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by awanlin — CHANGES_REQUESTED** [review]
This is an awesome first step on this subject @tcardonne, thanks for this contribution! 🚀 

Left a pair of comments.

I'd like to keep #20155 as I see this as the first step, I'd eventually like to see how to actually use the Traces and how to add more metrics. Hope that's alright with you?

Oh, also don't worry about the E2E tests failing, that's clearly not due to your changes 👍

File: docs/tutorials/setup-opentelemetry.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  instrumentations: [getNodeAutoInstrumentations()],
+});
+
+sdk.start();
+```
+
+In the `index.ts`, import this file **at the beginning**:
+
+```typescript
+import './instrumentation'; // Setup the OpenTelemetry instrumentation
```
[the remark this answers is not in the document]
Comment (tcardonne) [reply docs/tutorials/setup-opentelemetry.md:51]: I tested it on our own Backstage instance which uses the new backend system :+1:

[the remark this answers is not in the document]
Comment (awanlin) [reply docs/tutorials/setup-opentelemetry.md:51]: Awesome, thanks!

**Comment by tcardonne** [comment]
Hey @awanlin, thanks for the review!

> I'd like to keep #20155 open as I see this as the first step, I'd eventually like to see how to actually use the Traces and how to add more metrics. Hope that's alright with you?

Yes of course, there's still work to do in my opinion.
It would be nice to also have an exhaustive listing of available metrics per package/plugin. We just started to play with metrics on our side, maybe I'll hack a follow-up PR to add this list as a reference somewhere.

**Review by benjdlambert — APPROVED** [review]
Nice - this is great! Been wanting something like this for a while :tada:

**Review by awanlin — APPROVED** [review]

**Comment by awanlin** [comment]
@benjdlambert, can you merge this? It's a docs only change so the E2E tests are not related

**Comment by awanlin** [comment]
Thanks again @tcardonne and thanks for the merge @benjdlambert 🚀

**Pull request merged**